### ISSUE TITLE
Sort by Users Causes Issue Details page 'Replay' section Error 'The field 'user' is not a sortable field'

### ISSUE THREAD

**Issue by ReneGreen27** [getsentry/sentry#62784]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Go to Issues
2. Sort by Last Seen
3. Sort by Users
4. Open an Issue
5. Go to Replays
6. Error appears “The field `user` is not a sortable field.” . In dev tools, I can see Bad Request with GET

To note if navigating to the issue from my default Issues view, in which I have sort by Users enabled this error does not occur. 

Customer found and I was able to replicate this error when navigating to the issue from the Issues view, but first I had sort by Last seen enabled, and then I switched to sort by Users.

### Expected Result

Expected: Customer can view the replays.

### Actual Result

<img width="1240" alt="Screenshot 2024-01-08 at 1 16 43 PM" src="https://github.com/getsentry/sentry/assets/100724053/9b6f8997-7871-409d-aece-78d593f5e983">

### Product Area

Issues

**Comment by lobsterkatie** [comment]
Yup, definitely seems like a bug. I can replicate it, too. Thanks for reporting!

I'm adding this to our backlog to fix.

**Comment by MichaelSun48** [comment]
This no longer appears to be an issue in the new or old Issue Details UI. Going to close this since I can't reproduce this, but anyone feel free to reopen if you can reproduce it.
